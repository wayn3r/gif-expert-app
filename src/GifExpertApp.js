import { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'
import PropTypes from 'prop-types'

const GifExpertApp = ({ defaultCategories = [] }) => {
    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball']
    const [categories, setCategories] = useState(defaultCategories)
    const handleAdd = categorie => {
        setCategories([categorie, ...categories])
    }

    return (
        <div>
            <h2>GifExpertApp</h2>
            <hr />
            <AddCategory addCategory={handleAdd} />
            <ol>
                {categories.map(category => (
                    <GifGrid category={category} key={category} />
                ))}
            </ol>
        </div>
    )
}
GifExpertApp.propTypes = {
    defaultCategories: PropTypes.arrayOf(PropTypes.string),
}
export default GifExpertApp
