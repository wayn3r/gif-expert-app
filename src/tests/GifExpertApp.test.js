import { shallow } from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('Pruebas para el componente <GifExpertApp />', () => {
    const wrapper = shallow(<GifExpertApp />)
    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Dragon Ball']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
})
