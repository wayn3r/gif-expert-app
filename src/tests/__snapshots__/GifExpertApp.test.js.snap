// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente <GifExpertApp /> debe de mostrar el componente correctamente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    addCategory={[Function]}
  />
  <ol />
</div>
`;

exports[`Pruebas para el componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    addCategory={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;
